dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/EntryVersionHelper.java

```diff
@@ -258,10 +258,11 @@ default Map<String, ImmutablePair<SourceFile, FileDescription>> getSourceFiles(l
 
             versionDAO.enableNameFilter(finalTagName);
             T entry = getDAO().findById(workflowId);
-            checkOptionalAuthRead(user, entry);
+            checkExistsEntry(entry);
+            checkCanRead(user, entry);
 
             // tighten permissions for hosted tools and workflows
-            if (!user.isPresent() || AuthenticatedResourceInterface.userCannotRead(user.get(), entry)) {
+            if (!user.isPresent() || !isOwner(user.get(), entry)) {
```

Comment:
Yes, this code (both pre and post PR) disallows access to unpublished hosted workflow source files unless the user actually owns the files.  Is SAM supposed to figure into that equation,?

Reply:
> Yes, this code (both pre and post PR) disallows access to unpublished hosted workflow source files unless the user actually owns the files. Is SAM supposed to figure into that equation,?

Yes, if you have read or write permissions from SAM and are not an owner, you should be able to see a workflow's source files.

Yep, it looks like an existing bug, which I didn't realize until you pointed it out.

Reply:
I just looked in the staging UI, with a workflow Denis shared with me, and I can see the source files of a shared, not published, workflow without an error. I was confused why it worked, and looked at it -- the `getVersionsSourceFiles()` method also in this file, doesn't do a permissions check, but this `getSourceFiles()` method does. The UI, at least the page I'm looking at, is invoking an API that ends up invoking `getVersionsSourceFiles()`, which is why we don't get an error.

It would be nice to consolidate the methods, if possible, and if not, be consistent in how auth is handled.

Can be a separate PR (please create a ticket if so).

Reply:
Probably better to unify the methods in a separate PR.  New ticket: https://ucsc-cgl.atlassian.net/browse/SEAB-4625